src/main/scala/io/iohk/ethereum/blockchain/sync/LoadableBloomFilter.scala

```diff
@@ -0,0 +1,41 @@
+package io.iohk.ethereum.blockchain.sync
+
+import com.google.common.hash.{BloomFilter, Funnel}
+import io.iohk.ethereum.blockchain.sync.LoadableBloomFilter.BloomFilterLoadingResult
+import io.iohk.ethereum.db.dataSource.RocksDbDataSource.IterationError
+import monix.eval.Task
+import monix.reactive.{Consumer, Observable}
+
+class LoadableBloomFilter[A](bloomFilter: BloomFilter[A], source: Observable[Either[IterationError, A]]) {
+  val loadFromSource: Task[BloomFilterLoadingResult] = {
```

Comment:
I have mixed feelings about this setup TBH. If underlying bloom filter can be used only once loading completes - let's make a function which takes a source and returns task with ready-to-use bloom filter.

Reply:
for now we are using underlying bloom filter that way (first load then use), as it is simplest possible scenario and loading time for etc is relatively shot i.e 2-3 min.

As underlying bloom filter is thread safe, we could probably start loading filter from storage and at the same time start state sync, but i would do this as separate task. (It could be especially useful if we were to support ETH when loading nodes would take probably more that 10min which is probably not acceptable)

That why i would leave it as it is.